clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java

```diff
@@ -149,11 +156,14 @@ private void onSuccessfulResponse(final FindCoordinatorResponseData.Coordinator
                 coordinator.host(),
                 coordinator.port());
         log.info("Discovered group coordinator {}", coordinator);
-        coordinatorRequestState.reset();
+        coordinatorRequestState.onSuccessfulAttempt(currentTimeMs);
     }
 
-    private void onFailedCoordinatorResponse(final Exception exception, final long currentTimeMs) {
-        coordinatorRequestState.updateLastFailedAttempt(currentTimeMs);
+    private void onFailedResponse(
+        final long currentTimeMs,
+        final Throwable exception
+    ) {
+        coordinatorRequestState.onFailedAttempt(currentTimeMs);
```

Comment:
How about moving this to the end of the function as well, since `nonRetriableErrorHandler.handle(exception);` may take time?

Reply:
Hmm, I think it seems ok to log the time of the failed response. Any time spent in `nonRetriableErrorHandler.handle(exception)` can count toward the backoff. Does that make sense?

Reply:
Yeah that makes sense too. I was thinking on `onSuccessfulAttempt` we record the time in the end of the function so why not doing the same, but I think its does not matter much.

Reply:
True, but note the record time is provided as a parameters, so it doesn't actually matter. I was going to move this to the end of the line also, but it requires a little refactoring here due to the shortcut returns.